.github/workflows/docker.yaml

```diff
@@ -314,17 +314,24 @@ jobs:
       - name: Get canonical image name
         id: name
         run: echo "canonical=$(echo ${{ needs.image.outputs.images }} | sed 's/ .*//')" >> "$GITHUB_OUTPUT"
-      - name: Pull images
-        run: |
-          for ARCH in ${{ inputs.archs }} ; do
-            docker pull "${{ steps.name.outputs.canonical }}-${ARCH}"
-          done
       - name: Create and push manifest
         run: |
-          for ARCH in ${{ inputs.archs }} ; do
-            echo "${{ steps.name.outputs.canonical }}-${ARCH}"
-          done | \
-          xargs docker manifest create "${{ steps.name.outputs.canonical }}"
+          # Sometimes GCR gives random errors. Let's try a few times before giving up.
+          for RETRY in $(seq 10) ; do
+            for ARCH in ${{ inputs.archs }} ; do
+              if ! docker pull "${{ steps.name.outputs.canonical }}-${ARCH}" ; then
+                echo "Attempt #${RETRY} failed."
+                sleep 30
+                continue
+              fi
+            done
+            IMGS=$(echo ${{ inputs.archs }} | tr ' ' '\n' | sed 's%^%${{ steps.name.outputs.canonical }}-%' | tr '\n' ' ')
+            if docker manifest create "${{ steps.name.outputs.canonical }}" ${IMGS} ; then
+              break
+            fi
+            echo "Attempt #${RETRY} failed."
+            sleep 30
```

Comment:
If the manifest fails to get created, it will also re-do the `pull`ing, right? That shouldn't actually be a problem, but it may be noisy.

Reply:
Yeah. It's also possible that the manifest creation failure is somehow related to the pull doing something wonky. It shouldn't hurt, and there's a vague chance it'll help something.